tiledb/dataframe_.py

```diff
@@ -693,7 +693,15 @@ def _write_array(
     row_start_idx=None,
     timestamp=None,
 ):
+
     with tiledb.open(uri, "w", timestamp=timestamp) as A:
+        for j in range(A.schema.nattr):
+            attr = A.schema.attr(j)
+            if attr.enum_label is not None:
+                enmr = A.enum(attr.enum_label).values()
+                df[attr.name] = df[attr.name].cat.set_categories(enmr)
```

Comment:
We should probably do this at the arrow level eventually, this is going to be expensive. (sc-38461)